File: hybrid_jumping_robot/scripts/Actuator_robot.py
#!/usr/bin/env python3

import rospy
from std_msgs.msg import Float64


class Actuator:
    extension_publisher = ''
    extension_connection = '/hybrid_robotV0_2/extension_base_joint_position_controller/command'
    extension_position = 0.0

    def __init__(self, name):
        rospy.init_node(name, anonymous=True)
        rospy.Rate(5)
        self.extension_publisher = rospy.Publisher(self.extension_connection, Float64, queue_size=10)

    def set_extension_position(self, pos):
        rate = rospy.Rate(10)  # 10hz
        self.extension_publisher.publish(pos)
        rate.sleep()

File: hybrid_jumping_robot/src/Actuator_robot.cpp
#include "Actuator_robot.hpp"
#include <std_msgs/Float64.h>

Actuator::Actuator(): rate(10) {
  ROS_INFO("Actuator constructor");
  ros::NodeHandle nh;
  extension_publisher = nh.advertise<std_msgs::Float64>(extension_connection, 10);
}

void Actuator::set_extension_position(Float64 pos) {
  std_msgs::Float64 msg;
  msg.data = pos;
  extension_publisher.publish(msg);
  rate.sleep();
}